backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/controllers/EmailControllerTest.java

```diff
@@ -0,0 +1,92 @@
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Arrays;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles("autotests")
+public class EmailControllerTest {
+
+    private static final String BASE_URL = Mappings.PORT_AUTOTESTS + Mappings.API_VERSION + Mappings.EMAIL;
+    private MockMvc mockMvc;
+    private static Email email;
+    private static ObjectMapper mapper;
+
+    @Mock
+    private EmailService emailService;
+
+    @InjectMocks
+    private EmailController emailController;
+
+    @BeforeAll
+    public static void init() {
+        mapper = new ObjectMapper();
+        email = new Email();
+        email.setSubject("subject");
+        email.setMessageContext("message context");
+    }
+
+    @BeforeEach
+    public void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(emailController).build();
+        email.setSubject("subject");
+        email.setMessageContext("message context");
+    }
+
+    @Test
+    public void createEmailTest() throws Exception {
+        String jsonRequest = mapper.writeValueAsString(email);
+        mockMvc.perform(post(BASE_URL + Mappings.CREATE)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRequest))
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
+
+	@Test
+	public void messageCantBeEmpty() throws Exception {
+        email.setMessageContext("");
+        String jsonRequest = mapper.writeValueAsString(email);
+        mockMvc.perform(post(BASE_URL + Mappings.CREATE)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRequest))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.success", Matchers.is(false)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message", Matchers.contains("The message can't be empty")));
+    }
+
+    @Test
+    public void subjectCantBeEmpty() throws Exception {
+        email.setSubject("");
+        String jsonRequest = mapper.writeValueAsString(email);
+        mockMvc.perform(post(BASE_URL + Mappings.CREATE)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRequest))
+                .andDo(print())
+                .andExpect(status().isOk())
```

Comment:
When user does not provide all necessary data, backend should return BAD_REQUEST code.